.circleci/codebuild-checkout.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash -e
+
+# This script checks out a branch & loads git tags.
+
+git status
+
+echo "CODEBUILD_SOURCE_VERSION=$CODEBUILD_SOURCE_VERSION"
+echo "BRANCH_NAME=$BRANCH_NAME"
+echo "CODEBUILD_WEBHOOK_TRIGGER=$CODEBUILD_WEBHOOK_TRIGGER"
+
+if [[ "$CODEBUILD_SOURCE_VERSION" != "" ]]; then
+  CODEBUILD_SOURCE_VERSION_REF=$(git show-ref | grep $CODEBUILD_SOURCE_VERSION)
+  echo "CODEBUILD_SOURCE_VERSION_REF=$CODEBUILD_SOURCE_VERSION_REF"
+fi
+
+# Codebuild doesn't checkout the branch by default
+if [[ "$CODEBUILD_WEBHOOK_TRIGGER" =~ ^pr/ || "$CODEBUILD_SOURCE_VERSION" =~ ^pr/ || "$CODEBUILD_SOURCE_VERSION_REF" =~ refs/pull/[0-9]+/head$ ]]; then
+  # If we're in PR workflow create temporary local branch.
+  # We detect if we're in PR by looking for pr/<number> pattern in code build env variables
+  # or by checking if commit is matching refs/pull/<number>/head.
+  echo "Creating temporary local branch for PR build"
+  TEMP_BRANCH_NAME=$(cat /proc/sys/kernel/random/uuid)
```

Comment:
do we need delete the `TEMP_BRANCH_NAME` somewhere?

Reply:
no. it's not pushed to github.

Reply:
I'm using uuid, not arbitrary string in case name is already taken by existing branch.
this branch exists only locally and is then dropped, i.e. not pushed to git.